File: tests/rosettacode/bubblesort.clj
;; Bubble sort
(defn bubble-pass [coll]
  (reduce (fn [acc x]
            (if (and (seq acc) (> (peek acc) x))
              (conj (pop acc) x (peek acc))
              (conj acc x)))
          [] coll))

(defn bubble-sort [coll]
  (let [sorted (bubble-pass coll)]
    (if (= sorted coll) sorted (recur sorted))))

(let [arr [5 3 8 4 2 7 1 6]]
  (println "Before:" arr)
  (println "After: " (bubble-sort arr)))

File: tests/rosettacode/bubblesort.go
package main

import "fmt"

func bubbleSort(arr []int) {
	n := len(arr)
	for i := 0; i < n-1; i++ {
		for j := 0; j < n-1-i; j++ {
			if arr[j] > arr[j+1] {
				arr[j], arr[j+1] = arr[j+1], arr[j]
			}
		}
	}
}

func main() {
	arr := []int{5, 3, 8, 4, 2, 7, 1, 6}
	fmt.Println("Before:", arr)
	bubbleSort(arr)
	fmt.Println("After: ", arr)
}
